### ISSUE TITLE
No code-splitting without existing .cache

### ISSUE THREAD

**Issue by elliotdickison** [gatsbyjs/gatsby#19215]
###  Description

When I run `gatsby build` in my project without a pre-existing gatsby `.cache` (e.g. on a fresh clone of the project, or when the command is run on my CI server), the app is not code-split at the page level (there is one giant JS bundle for the entire site). When I run `gatsby build` a second time (after `.cache` has been created), page-level code-splitting works as expected.

### Steps to reproduce

The basic commands to reproduce though are:

1. `rm -rf .cache`
2. `gatsby build` (code-splitting does not work)
3. `gatsby build` (code-splitting works)

### Expected result

I would expect that `gatsby build` would code-split regardless of the presence of a pre-existing `.cache` directory (and be deterministic in general).

### Actual result

I get one giant JS bundle at `public/app-<hash>.js` on the first cache-less build. The resulting site still functions - it just loads a whopping 2.8MB bundle containing *everything* when the site is initially loaded (we have a very large site with many pages). There are no page-level js files. The `public/chunk-map.json` file only lists the app bundle: `{"app":["/app-<hash>.js"]}`. Interestingly, if I dig down into `public/page-data` the json in there does reference page-level js chunks. For example in `public/page-data/faq/page-data.json` I see `{"componentChunkName":"component---src-pages-faq-js","path":"/faq/","data":...}`.

### Environment

```
  System:
    OS: macOS 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 12.9.0 - ~/.nvm/versions/node/v12.9.0/bin/node
    npm: 6.10.2 - ~/.nvm/versions/node/v12.9.0/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 78.0.3904.70
    Firefox: 68.2.0
    Safari: 13.0.3
  npmPackages:
    gatsby: ^2.17.0 => 2.17.0
    gatsby-link: ^2.2.22 => 2.2.22
    gatsby-plugin-emotion: ^2.0.7 => 2.0.7
    gatsby-plugin-layout: ^1.1.13 => 1.1.13
    gatsby-plugin-react-helmet: ^3.1.13 => 3.1.13
    gatsby-plugin-webpack-bundle-analyzer: ^1.0.5 => 1.0.5
    gatsby-source-filesystem: ^2.1.33 => 2.1.33
    gatsby-transformer-json: ^2.2.15 => 2.2.15
    gatsby-transformer-react-docgen: ^4.0.1 => 4.1.3
```

P.S. Unfortunately this is a large internal project and I'm not able to share a repo. If there are particular config files that it would be helpful to share I probably can. I understand I'm not providing a ton to go on and this may be related to some non-standard thing I'm doing. I'd appreciate any pointers to help me debug this myself though. Thanks for the help and for Gatsby, I love using it!

**Comment by m-allanson** [comment]
How strange. Do you get the same behaviour if you replace `rm -rf .cache` with `gatsby clean`?

**Comment by elliotdickison** [comment]
This resolved itself with a seemingly unrelated change. We did not make any Gatsby/Webpack/Babel configuration changes or change any Gatsby dependencies, but all of a sudden the issue went away with a particular commit to our app code (after 30 or 40 commits where this was an issue). The only thing in the commit that seemed like it could be related was that we migrated from [react-loadable](https://www.npmjs.com/package/react-loadable) to [@loadable/component](https://www.npmjs.com/package/@loadable/component) for lazy loading a few heavy bits of code. However, we had been using @loadable/component in the past and this code-splitting issue was still a problem. Closing since this is no longer an issue for us 🤷‍♂️.
